### DISCUSSION TITLE
How do I read/inspect an attestation stored in an OCI registry

### DISCUSSION THREAD

**Discussion by rafaeltuelho** [sigstore/cosign#3547] [Q&A] [answered]
I have my attestation stored as an OCI artifact in a Registry. How do I read/inspect or verify against it using the CLI?

**Comment by adilburaksen** [comment, marked as answer]
Two commands depending on what you want to do:

**Download and read the attestation payload:**
```sh
cosign download attestation <image>:<tag>
```
This outputs the DSSE envelope(s) as newline-delimited JSON. The actual predicate is base64-encoded in the `payload` field:
```sh
cosign download attestation <image>:<tag> | jq -r '.payload' | base64 -d | jq .
```

**Verify and read in one step:**
```sh
cosign verify-attestation   --type <predicate-type>   --certificate-identity <identity>   --certificate-oidc-issuer <issuer>   <image>:<tag>
```
This verifies the signature and prints the verified payload to stdout. Common predicate types: `slsaprovenance`, `cyclonedx`, `spdx`, `vuln`, `custom`.

**For a bundle stored in a file instead of a registry:**
```sh
cosign verify-attestation --bundle <bundle.json> --type <type> <image>
```

If you just want to inspect without verifying (not recommended for production), `cosign download attestation` is enough — it fetches the raw OCI artifact and decodes the envelope.
